From 32dab08569a20529bd1d0ac4a3aa1a1082ef1a98 Mon Sep 17 00:00:00 2001
From: ObjectAscended <93239187+ObjectAscended@users.noreply.github.com>
Date: Wed, 18 Mar 2026 09:08:58 +0000
Subject: [PATCH] Bypass license verification in DoLicenseCheck

Refactors the license check logic to skip JWT decoding and cryptographic signature verification, instead directly saving and returning the license token.
---
 .../domain/usecases/DoLicenseCheck.java            | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/domain/src/main/java/org/cryptomator/domain/usecases/DoLicenseCheck.java b/domain/src/main/java/org/cryptomator/domain/usecases/DoLicenseCheck.java
index 8a8cc9e5..74b126e3 100644
--- a/domain/src/main/java/org/cryptomator/domain/usecases/DoLicenseCheck.java
+++ b/domain/src/main/java/org/cryptomator/domain/usecases/DoLicenseCheck.java
@@ -33,18 +33,8 @@ public class DoLicenseCheck {
 
 	public LicenseCheck execute() throws BackendException {
 		license = useLicenseOrRetrieveFromPreferences(license);
-		try {
-			DecodedJWT jwt = AndroidLicenseVerifier.verify(license, AndroidLicenseVerifier.ANDROID_PUB_KEY);
-			sharedPreferencesHandler.setLicenseToken(license);
-			return jwt::getSubject;
-		} catch (SignatureVerificationException | JWTDecodeException | FatalBackendException e) {
-			if (e instanceof SignatureVerificationException && isDesktopSupporterCertificate(license)) {
-				throw new DesktopSupporterCertificateException(license);
-			}
-			throw new LicenseNotValidException(license);
-		} catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
-			throw new FatalBackendException(e);
-		}
+		sharedPreferencesHandler.setLicenseToken(license);
+		return () -> license;
 	}
 
 	private String useLicenseOrRetrieveFromPreferences(String license) throws NoLicenseAvailableException {
-- 
2.53.0

